### ISSUE TITLE
Code Contribution: 【Easy】【Operator Development】channel_shuffle

### ISSUE THREAD

**Issue by kiddyjinjin** [FlagOpen/FlagGems#1122]
## Description 任务介绍

Develop forward function for channel_shuffle operator.  
开发 channel_shuffle 算子的前向功能。

此算子属于 layout 类算子。

## Requirements 任务要求

__Interface 接口__

`channel_shuffle(Tensor self, SymInt groups) -> Tensor`

__Function reference 功能参考__

PyTorch：  
https://pytorch.org/docs/stable/generated/torch.nn.ChannelShuffle.html

实现逻辑类似于 shuffle channels：  
输入 `(N, C, H, W)`，将 C 按 groups 分组，然后交换 group 和 subchannel 维度。

__Implementation reference 实现参考__

FlagGems layout 示例参考：  
https://github.com/FlagOpen/FlagGems/blob/master/src/flag_gems/ops/permute.py

__Behavior 行为要求__

- 支持输入形状 `(N, C, ...)`，并要求 `C % groups == 0`
- 实现步骤应与 PyTorch 对齐：
  - view → reshape → transpose → reshape 回原状结构
- 不涉及复杂数值计算，属于纯 layout 变换

## DDL 提交时间

两周内提交 PR，并附正确性测试与性能测试。

**Comment by kiddyjinjin** [comment]
@ridethepig  认领

**Comment by bin913** [comment]
The latest repository already has the implementation of this operator.
